libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/KotlinNativeTargetConfigurator.kt

```diff
@@ -272,12 +273,14 @@ open class KotlinNativeTargetConfigurator<T : KotlinNativeTarget>(
 
         // Create an aggregate link task for each compilation.
         target.compilations.all {
-            project.tasks.create(it.binariesTaskName)
+            project.registerTask<DefaultTask>(it.binariesTaskName)
         }
 
         project.whenEvaluated {
-            target.binaries.forEach {
-                project.tasks.getByName(it.compilation.binariesTaskName).dependsOn(it.linkTaskName)
+            target.binaries.forEach { binary ->
+                project.tasks.named(binary.compilation.binariesTaskName).configure { binariesTask ->
+                    binariesTask.dependsOn(binary.linkTaskName)
```

Comment:
don't you need a `named(binary.linkTaskName)` ?

Reply:
Not needed, Gradle allows taking just the name of the task as a `String`.